src/mainviews/detail-view/detail-view.tsx

```diff
@@ -63,17 +63,40 @@ export function DetailView(props: IDetailViewProps): JSX.Element {
   const ss = new SchedulerService({});
 
   const fetchJobDetailModel = async () => {
-    const jobFromService = await ss.getJob(props.model.id);
-    const jobDetailModel = convertDescribeJobtoJobDetail(jobFromService);
-    setJobsModel(jobDetailModel);
+    try {
+      const jobFromService = await ss.getJob(props.model.id);
+      const jobDetailModel = convertDescribeJobtoJobDetail(jobFromService);
+      setJobsModel(jobDetailModel);
+    } catch (e: any) {
+      setJobsModel({
+        jobId: '',
+        jobName: '',
+        inputFile: '',
+        environment: '',
+        job_files: [],
+        downloaded: false,
+        createError: e.message
+      });
+    }
   };
 
   const fetchJobDefinitionModel = async () => {
-    const definitionFromService = await ss.getJobDefinition(props.model.id);
-    const jobDefinitionModel = convertDescribeDefinitiontoDefinition(
-      definitionFromService
-    );
-    setJobDefinitionModel(jobDefinitionModel);
+    try {
+      const definitionFromService = await ss.getJobDefinition(props.model.id);
+      const jobDefinitionModel = convertDescribeDefinitiontoDefinition(
+        definitionFromService
+      );
+      setJobDefinitionModel(jobDefinitionModel);
+    } catch (e: any) {
+      setJobDefinitionModel({
+        definitionId: '',
+        name: '',
+        inputFile: '',
+        outputPath: '',
+        environment: '',
+        createError: e.message
+      });
+    }
   };
```

Comment:
How about instead of setting this on the model, we keep the state local? The model should only be defined in the case where we successfully fetched a job / job definition from the backend, and we should keep it null otherwise. This would mean we need a separate state variable that is maintained as follows:

```tsx
const [fetchError, setFetchError] = useState<string>();

...

try {
  ...
} catch (e: Error) {
  setFetchError(e.message);
}
```

And right before rendering the body of the page, we display the error banner if necessary:

```tsx
return (
  ...
  {fetchError && <Alert severity="error">{fetchError}</Alert>}
  {props.jobsView === JobsView.JobDetail && jobModel && (
     ...
   )}
  ...
)
```

Finally, you should revert your changes to job definition and job detail.

Reply:
Just pushed a commit that implements my suggestions.